src/txmempool.cpp

```diff
@@ -677,14 +677,18 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
     assert(innerUsage == cachedInnerUsage);
 }
 
-void CTxMemPool::queryHashes(vector<uint256>& vtxid)
+void CTxMemPool::queryHashes(vector<uint256>& vtxid, int64_t mintime, int64_t maxtime)
 {
     vtxid.clear();
 
     LOCK(cs);
     vtxid.reserve(mapTx.size());
-    for (indexed_transaction_set::iterator mi = mapTx.begin(); mi != mapTx.end(); ++mi)
-        vtxid.push_back(mi->GetTx().GetHash());
+    //Doesn't use the time sorted index to avoid an information leak about when a transaction entered the mempool.
```

Comment:
Right the order is the problem. It's true that it could sort, but thats another expensive operation, and the result being in fee sorted order is quite useful.